cec-mini-kb.cpp

```diff
@@ -210,6 +225,19 @@ int main(int argc, char* argv[])
 		if ((arg == "-h") || (arg == "--help")) {
 			show_usage(argv[0]);
 			return 0;
+		} else if ((arg == "-a") || (arg == "--adapter")) {
+			if (i + 1 < argc) {
+				i++;
+				if (legal_int(argv[i])){
+					adapter_number = atoi(argv[i]);
+				} else {
+					std::cerr << "--adapter option requires a integer number argument." << std::endl;
+					return 1;
+				}
```

Comment:
Max number of devices is already hardcoded here as 10: https://github.com/spleen1981/cec-mini-kb/blob/be4289751bd80470c33847073a790b83356696db/cec-mini-kb.cpp#L259
Hence we can just test the single char and simplify as follows:
```
				adapter_number=atoi(*argv[i]);
				if ( *argv[i] != '0' && !adapter_number){
					std::cerr << "--adapter option requires a integer number argument between 0 and 9." << std::endl;
					return 1;
				}
```

Reply:
Yes, it really would be simpler this way, igualmente me daba un error llamar a atoi de la manera que usted indica, asi que lo cambio por esto
```
				adapter_number = atoi(argv[i]);
				if ( adapter_number < 0 || adapter_number > 9) {
					std::cerr << "--adapter option requires a integer number argument between 0 and 9." << std::endl;
					return 1;
				}
```

Reply:
I better add one more check in case atoi returns zero for a string that is not an integer
```
				adapter_number = atoi(argv[i]);
				if ( ( adapter_number == 0 && strcmp(argv[i],"0") != 0 ) || adapter_number < 0 || adapter_number > 9) {
					std::cerr << "--adapter option requires a integer number argument between 0 and 9." << std::endl;
					return 1;
				}
```